apis/container/v1beta2/cluster_types.go

```diff
@@ -1265,6 +1268,27 @@ type DefaultSnatStatus struct {
 	Disabled bool `json:"disabled"`
 }
 
+// DnsConfig contains the desired set of options for configuring clusterDNS.
+type DnsConfig struct {
+	// ClusterDns indicates which in-cluster DNS provider should be used.
+	// Possible values:
+	//   "PROVIDER_UNSPECIFIED" - indicates the default value.
+	//   "PLATFORM_DEFAULT" - indicates using the GKE default DNS
+	//  provider(kube-dns) for DNS resolution.
+	//   "CLOUD_DNS" - indicates using CloudDNS for DNS resolution.
+	ClusterDns string `json:"clusterDns,omitempty"`
+
+	// ClusterDnsScope indicates the scope of access to cluster DNS records.
+	// Possible Values:
+	//   "DNS_SCOPE_UNSPECIFIED" - indicates the default value,
+	//  will be inferred as cluster scope.
+	//   "VPC_SCOPE" - indicates that DNS records are accessible from within the VPC.
+	ClusterDnsScope string `json:"clusterDnsScope,omitempty"`
+
+	// ClusterDnsDomain is the suffix used for all cluster service records.
+	ClusterDnsDomain string `json:"clusterDnsDomain,omitempty"`
+}
+
```

Comment:
Same as before, all these fields are optional, so we need to add the `// +optional` marker.

Also, when a field is optional, its type should be a pointer. So these 3 fields should be changed from `string` to `*string`.

Reply:
The `ClusterDns` and `ClusterDnsScope` fields could also use the kubebuilder marker `+kubebuilder:validation:Enum` to validate its possible values. 

Check [this example](https://github.com/crossplane-contrib/provider-gcp/blob/89e8a848bc7164a59c0a5f6bff61bb1b4354e33e/apis/cache/v1beta1/cloudmemorystore_instance_types.go#L126) that is using this validation.